src/hooks/spaces/useSpaceBySlug.ts

```diff
@@ -0,0 +1,63 @@
+import { useState } from "react";
+import { useFirestore, useFirestoreCollectionData } from "reactfire";
+import Bugsnag from "@bugsnag/js";
+
+import { AnyVenue } from "types/venues";
+
+import { withIdConverter } from "utils/converters";
+import { WithId } from "utils/id";
+
+type UseSpaceBySlugResult = {
+  space?: WithId<AnyVenue>;
+  isLoaded: boolean;
+};
+
+/**
+ * Hook which will return the space when the slug is provided.
+ * The intention is to be used on the client side, when the space slug is provided in the url.
+ * @param spaceSlug
+ * @returns
+ */
+export const useSpaceBySlug: (spaceSlug?: string) => UseSpaceBySlugResult = (
+  spaceSlug
+) => {
+  const [error, setError] = useState<string>();
+  const firestore = useFirestore();
+
+  const spacesRef = firestore
+    .collection("venues")
+    .where("slug", "==", spaceSlug ?? "")
+    .withConverter(withIdConverter<AnyVenue>());
+
+  const { data: spaces, status } = useFirestoreCollectionData<WithId<AnyVenue>>(
+    spacesRef,
+    {
+      initialData: undefined,
+    }
+  );
+
+  const isSpaceLoaded = status !== "loading";
+
+  if (!spaces?.[0]) {
+    setError(`Space with the following slug: ${spaceSlug} does not exist.`);
```

Comment:
```suggestion
    return {
      space: spaces?.[0],
      isLoaded: isSpaceLoaded,
      error: `Space with the following slug: ${spaceSlug} does not exist.`,
  };
```

Reply:
Fixed in: https://github.com/sparkletown/sparkle/pull/2548/commits/9cc14823a26b81030e32d035218556fed2c2c758